### PR TITLE
Re-install libtcmu API and header files

### PR DESCRIPTION
**Pull request by baihuahua**
Thanks to Mike and other persons' great efforts on re-licenseing tcmu-runner/libtcmu now we can link qemu-tcmu and libtcmu. It's time to bring libtcmu API back. This patchset reverts two commits to re-install libtcmu API and tunes header files for dependence. Now when we compile tcmu-runner, libtcmu API and four headers(libtcmu.h, libtcmu_common.h, libtcmu_priv.h, darray.h) will be installed. With this patchset qemu-tcmu can be successfully compiled and works pretty properly. @amarts @lxbsz 
We also can base on these patches to tune libtcmu further and make libtcmu more suitable for different  TCMU daemons.

### CODE REVIEW

File: libtcmu_priv.h [hunk 1]
```diff
@@ -20,11 +20,7 @@
 #include <gio/gio.h>
 #include <pthread.h>
 
-#include "scsi_defs.h"
 #include "darray.h"
-#include "ccan/list/list.h"
-#include "tcmur_aio.h"
-#include "tcmu-runner.h"
```
Comment (mikechristie) [inline libtcmu_priv.h]: This breaks the build. Doing a make results in this error:

> /root/targetcli/devel/tcmu-runner/api.c: In function ‘tcmu_emulate_evpd_inquiry’:
> /root/targetcli/devel/tcmu-runner/api.c:627:15: error: dereferencing pointer to incomplete type
>    if (rhandler->unmap) {
>                ^
> /root/targetcli/devel/tcmu-runner/api.c:724:15: error: dereferencing pointer to incomplete type
>    if (rhandler->unmap)
> 

I think you need to move some more stuff around like add a include tcmu-runner.h somewhere. Or maybe drop this patch until Xiubo has done his surgery on the code to separate/move the lib stuff and tcmu-runner stuff.

Comment (baihuahua) [reply libtcmu_priv.h]: OK, Xiubo will finish his work very soon so drop this patch and wait for his PR.

Comment (mikechristie) [reply libtcmu_priv.h]: Oh wait. I think there was a small misunderstanding. I was just asking you to drop that one patch.

I think the other patches are ok.

Comment (baihuahua) [reply libtcmu_priv.h]: Yeh I misunderstood it but that's ok, these patches have been merged together with Xiubo's change and he opens a new PR. :-)

**Pull request closed without merging**